src/main/java/org/osiam/resources/scim/Meta.java

```diff
@@ -0,0 +1,320 @@
+                @JsonProperty("lastModified") @JsonSerialize(using = JsonDateSerializer.class) Date lastModified,
+                @JsonProperty("location") String location,
+                @JsonProperty("version") String version,
+                @JsonProperty("attributes") Set<String> attributes,
+                @JsonProperty("resourceType") String resourceType) {
+        this.created = created;
+        this.lastModified = lastModified;
+        this.location = location;
+        this.version = version;
+        this.attributes = attributes;
+        this.resourceType = resourceType;
+    }
+
+    private Meta(Builder builder) {
+        this.created = builder.created;
+        this.lastModified = builder.lastModified;
+        this.attributes = builder.attributes;
+        this.location = builder.location;
+        this.version = builder.version;
+        this.resourceType = builder.resourceType;
+    }
+
+    /**
+     * Gets the URI of the Resource being returned.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-5">SCIM core schema 2.0, section 5</a>
+     * </p>
+     *
+     * @return the location
+     */
+    public String getLocation() {
+        return location;
+    }
+
+    /**
+     * Gets the version of the Resource being returned.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-5">SCIM core schema 2.0, section 5</a>
+     * </p>
+     *
+     * @return the version
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * Gets the attributes to be deleted from the Resource
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-5">SCIM core schema 2.0, section 5</a>
+     * </p>
+     *
+     * @return a set of attributes to be deleted
+     */
+    public Set<String> getAttributes() {
+        return attributes;
+    }
+
+    /**
+     * Gets the date when the {@link Resource} was created
+     *
+     * @return the creation date
+     */
+    public Date getCreated() {
+        if (created != null) {
+            return new Date(created.getTime());
+        }
+        return null;
+    }
+
+    /**
+     * Gets the date when the {@link Resource} was last modified
+     *
+     * @return the last modified date
+     */
+    public Date getLastModified() {
+        if (lastModified != null) {
+            return new Date(lastModified.getTime());
+        }
+        return null;
+    }
+
+    /**
+     * Gets the type of the Resource (User or Group)
+     *
+     * @return the type of the actual resource
+     */
+    public String getResourceType() {
+        return resourceType;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method hashCode() has an NPath complexity of 15625](https://www.codacy.com/app/OSIAM/connector4java/file/1980681110/issues/source?bid=3081107&fileBranchId=3117155#l153)